.github/pull_request_template.md

```diff
@@ -6,22 +6,22 @@ change me!
 
 ### User-facing documentation
 
-(*must be* 2 items and both *must be* checked)
-<!-- Remove conflicting items that won't be checked. -->
+(*must be 2 checked* items, remove items that don't apply)
```

Comment:
How about splitting this up in two subsections, one for release notes, the other for docs?
In that case this line (per subsection) could be changed to something like `one item needs to be checked, remove items that don't apply`.

Would potentially make it easier to grasp that the structure is simply:

[ ] I took care of D
[ ] I requested changes for D
[ ] changes for D are not needed

where D is either release notes or official docs.

Reply:
@rukletsov idea was to have less template and more signal.

I thought about what you suggest but it goes in the opposite direction. Nevertheless, I applied it in the recent revision. Please have a look at [the preview](https://github.com/stackrox/stackrox/blob/misha/improve-pr-template/.github/pull_request_template.md) and let know what you think.

Please also don't hesitate to create suggestions using GitHub feature
![image](https://github.com/stackrox/stackrox/assets/537715/343f8f41-678e-4154-8d83-0f50d4913146)
this way, we can converge in less turns because this PR is all about small details and the actual markdown is the best way to convey the message.